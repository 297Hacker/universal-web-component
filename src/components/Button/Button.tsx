import React, {useRef, useEffect} from 'react';
import styled from 'styled-components';

type ButtonProps = {
    addClass?: string
    myClass?: string
    name?: string
    children?: string
    value?: string
    kind?: string[]
    onClick?: any
    onChange?: any
    type?: string
}

const MainButton = styled.button`
  background-color: #82BFE0;
  border-radius: 2px;
  border: none;
  color: #FFF;
  margin: 0 10px;
  padding: 10px 40px;
    &:hover {
        background-color: #000;
        cursor: pointer;
    }
`;

const Span = styled.span`
    display: none;
    
    &.view {
    display: block;
    }
`;

const Button: React.FC<ButtonProps> = (props) => {
    const focusButton:any = useRef(null);

    useEffect(() => {
        console.log('component is being rerendered');
    }, []);

    return <>
            <MainButton ref={focusButton}
                        className={props.myClass}
                        name={props.name}
                        value={props.value}
                        onClick={props.onClick}
                        onChange={props.onChange}>
                {props.children}
            </MainButton>
            <Span className={props.myClass}>view content when button is clicked</Span>
    </>
};

Button.defaultProps = {
    myClass: 'button',
    name: 'button',
    value: 'Click',
    children: 'Click button'
};

export default Button;