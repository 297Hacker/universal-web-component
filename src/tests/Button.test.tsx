import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../components/Button/Button';

configure({adapter: new Adapter()});

describe('<Button/>', () => {
    it("should render the button conent", () => {
        const component = shallow(<Button />);
        expect(component).toMatchSnapshot();
    });
});