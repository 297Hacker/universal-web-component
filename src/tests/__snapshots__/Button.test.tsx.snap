// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render the button conent 1`] = `
<Fragment>
  <styled.button
    className="button"
    name="button"
    value="Click"
  >
    Click button
  </styled.button>
  <styled.span
    className="button"
  >
    view content when button is clicked
  </styled.span>
</Fragment>
`;
